modules/apps/calendar/calendar-web/src/main/java/com/liferay/calendar/web/internal/upgrade/v_1_2_0/UpgradeStagingGroupTypeSettings.java

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.calendar.web.internal.upgrade.v_1_2_0;
+
+import com.liferay.petra.string.StringBundler;
+import com.liferay.portal.kernel.dao.jdbc.AutoBatchPreparedStatementUtil;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * @author Balázs Sáfrány-Kovalik
+ */
+public class UpgradeStagingGroupTypeSettings extends UpgradeProcess {
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		_removeStagedCalendarInstanceId();
+	}
+
+	private String _getNewTypeSettings(String typeSettings) {
+		Pattern pattern = Pattern.compile(
+			_STAGED_CALENDAR_PORTLET + _INSTANCE_SEPARATOR + "\\w{12}",
+			Pattern.MULTILINE);
+
+		Matcher matcher = pattern.matcher(typeSettings);
+
+		return matcher.replaceAll(_STAGED_CALENDAR_PORTLET);
+	}
+
+	private void _removeStagedCalendarInstanceId() throws Exception {
+		StringBundler sb = new StringBundler(5);
+
+		sb.append("select groupId, typeSettings from Group_ where ");
```

Comment:
@balazssk  we can't execute direct SQL for tables which don't belong to the module. That's the golden rule::
https://grow.liferay.com/share/DXP-+How+to+code+upgrade+process+and+backport+rules

Reply:
oops, guess creating something similar to the non-admin->admin upgrade process can be done here as well.